src/audio/eq_iir/eq_iir_ipc3.c

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2017-2022 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+//         Liam Girdwood <liam.r.girdwood@linux.intel.com>
+//         Keyon Jie <yang.jie@linux.intel.com>
+
+#include "eq_iir.h"
+#include <sof/audio/component.h>
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/data_blob.h>
+#include <sof/audio/buffer.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/ipc-config.h>
+#include <sof/common.h>
+#include <rtos/panic.h>
+#include <sof/ipc/msg.h>
+#include <rtos/alloc.h>
+#include <rtos/init.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/math/iir_df1.h>
+#include <sof/platform.h>
+#include <rtos/string.h>
+#include <sof/ut.h>
+#include <sof/trace/trace.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <user/eq.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+LOG_MODULE_DECLARE(eq_iir, CONFIG_SOF_LOG_LEVEL);
+
+#if CONFIG_FORMAT_S32LE && CONFIG_FORMAT_S16LE
```

Comment:
You could add comment that format conversion capability is for now only in IPC3 version, if that was the purpose of having these functions here.

Reply:
move them to eq_iir_ipc3.c means this is for IPC3 only.

Reply:
Yep, this is OK since the component capabilities are different for ipc3 and ipc4.